### PR TITLE
Join node in reduce mode doesn't keep existing msg properties

### PR DESCRIPTION
**Pull request by dceejay**
- [x] Bugfix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)

## Proposed changes

see https://discourse.nodered.org/t/no-response-object-after-split-join/63919

When in reduce mode the join node currently creates a new msg rather than re-using the existing one - which then drops things like any msg.res from a http request. This patch tries to fix that.

## Checklist

- [x] I have read the [contribution guidelines](https://github.com/node-red/node-red/blob/master/CONTRIBUTING.md)
- [x] For non-bugfix PRs, I have discussed this change on the forum/slack team.
- [x] I have run `grunt` to verify the unit tests pass
- [ ] I have added suitable unit tests to cover the new/changed functionality

### CODE REVIEW

**Review by Steve-Mcl — APPROVED** [review]
Looks good to me

**Comment by knolleary** [comment]
Should be back port this to 2.x?

**Comment by dceejay** [comment]
I guess ideally yes. But I'm offline until Monday.

**Comment by knolleary** [comment]
No problem - just wanted to get it labelled so it doesn't get missed

File: packages/node_modules/@node-red/nodes/core/sequence/17-split.js [hunk 1]
```diff
@@ -314,11 +314,13 @@ module.exports = function(RED) {
                         if (err) {
                             return done(err);
                         }
-                        msgInfo.send({payload: result});
+                        msgInfo.msg.payload = result;
+                        msgInfo.send(msgInfo.msg);
```
Comment (knolleary) [inline packages/node_modules/@node-red/nodes/core/sequence/17-split.js:318]: If we're now reusing `msgInfo.msg` should this get cloned before being sent?

Comment (Steve-Mcl) [reply packages/node_modules/@node-red/nodes/core/sequence/17-split.js:318]: As there is one input and one output in reduce mode, is cloning necessary?

Comment (dceejay) [reply packages/node_modules/@node-red/nodes/core/sequence/17-split.js:318]: I think you are right. As the reduce only happens when the "join" is complete then nothing should be able to change in that group of data after that point.

**Pull request merged**